cmd/backup-operator/README.md

```diff
@@ -0,0 +1,83 @@
+# etcd backup operator
+
+## Overview
+
+etcd backup operator backups the data of a etcd cluster running on [Kubernetes][Kube] to a remote storage such as AWS [S3][s3].
+
+## Getting Started
+
+Try out etcd backup operator by running it on Kubernetes and then create a `EtcdBackup` Custom Resource which contains the targeting etcd cluster and S3 backup config; the etcd backup operator automatically picks up the `EtcdBackup` Custom Resource, retrieves etcd snapshot, and then saves it to S3.
+>Note: The demo uses the `default` namespace.
+
+Prerequisites: 
+* access to a Kubernetes environment.
+* a running etcd cluster named `example-etcd-cluster`. See [Instruction][etcd_cluster_deploy].
+
+### Deploy etcd restore operator
+
+Once `example-etcd-cluster` is running, let's create a backup for `example-etcd-cluster` using etcd backup operator. 
+
+First, deploy an etcd backup operator:
+> Note: etcd backup operator also creates EtcdBackup CRD when starting.
+
+```sh
+$ kubectl create -f example/etcd-backup-operator/deployment.yaml
+$ kubectl get pod
+NAME                                    READY     STATUS    RESTARTS   AGE
+etcd-backup-operator-1102130733-hhgt7   1/1       Running   0          3s
+```
+
+### Create AWS Secret
+
+Then, create an Kubernetes secret that contains aws config/credential; etcd operator uses the secret to gain access to S3 in order to save the etcd snapshot.
+
+Verify that the local aws config and credentials files exist:
+```sh
+$ cat $AWS_DIR/credentials
+[default]
+aws_access_key_id = XXX
+aws_secret_access_key = XXX
+
+$ cat $AWS_DIR/config
+[default]
+region = <region>
+```
+
+Create kubernetes secret `aws`:
+
+`kubectl create secret generic aws --from-file=$AWS_DIR/credentials --from-file=$AWS_DIR/config`
+
+### Create Backup CR
+
+Fill the template `example/etcd-backup-operator/backup_cr.yaml` with concrete `s3Bucket` and `awsSecret` values and trigger a backup:
+
+```sh
+sed -e 's/<s3-bucket-name>/mybucket/g' \
+-e 's/<aws-secret>/aws/g' \
```

Comment:
Can you indent lines other than the first line?

Reply:
done